### DISCUSSION TITLE
Newline in table cell?

### DISCUSSION THREAD

**Discussion by mbolder** [zauberzeug/nicegui#1458] [Q&A] [answered]
### Question

Hello,
I am trying to display an address in a single table cell like

Mr B.
Test Str.12
New York

But new lines in the string don’t seem to work? Is there another way?

**Comment by falkoschindler** [comment]
Hi @mbolder!
Can you, please, provide a minimum reproducible code example of what you already tried?
This would allow us to help more efficiently. Thanks!

**Comment by mbolder** [comment]
That's my simplified attempt. The field is filled with the data, but in a single row. 

```py
columnsorder = [
            {
                "name": "shippadd",
                "label": "Ship",
                "field": "shippadd",
                "required": True,
                "align": "left",
                "sortable": False,
            },
]
roworder = []
name = "Mr.B"
street = "Test Str.12"
city = "New York"

ship = {}
shipadd = name+"<BR>"+street+"<BR>"+city
ship["shippadd"]=shipadd

roworder.append(ship)

tableord = ui.table(
                    columns=columnsorder, rows=roworder, row_key="shipadd", pagination=10
                ).classes("w-720")
```

**Comment by falkoschindler** [comment, marked as answer]
Here is a possible solution using scoped slots:

```py
table = ui.table(columns=[{'name': 'ship', 'label': 'Ship', 'field': 'ship', 'align': 'left'}],
                 rows=[{'ship': 'Mr. B<br>New York'}])
table.add_slot('body-cell-ship', '<q-td :props="props" v-html="props.value"></q-td>')
```

This instructs Vue to interpret the cell content as HTML rather than converting it into plain text.

**Reply by mbolder** [reply]
Brilliant !! First class help.

**Comment by retsyo** [comment]
[CrystalWindSnake](https://github.com/CrystalWindSnake) points out that the simplest way is to add this code at the beginning

```python
ui.add_head_html(
    r"""
<style>
    .q-table tbody td{
         white-space: pre
    }
</style>
"""
)
```
